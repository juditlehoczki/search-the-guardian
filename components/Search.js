import React from "react";
import { View, Text, TextInput, TouchableOpacity } from "react-native";
import s from "../styles.js";

const Search = (props) => {
  const { handleChange, searchInput, getArticles } = props;
  return (
    <View style={s.inputContainer}>
      <Text
        style={{
          alignSelf: "center",
          fontSize: 20,
          marginLeft: 5,
        }}
      >
        🔍
      </Text>
      <TextInput
        style={s.inputField}
        value={searchInput}
        placeholder="Search..."
        onChangeText={(value) => {
          handleChange(value);
        }}
      />
      <TouchableOpacity
        style={s.button}
        onPress={() => {
          searchInput.length > 0 && getArticles(1, searchInput);
        }}
      >
        <Text
          style={{
            color: "white",
            fontWeight: "bold",
          }}
        >
          Go!
        </Text>
      </TouchableOpacity>
    </View>
  );
};

export default Search;
