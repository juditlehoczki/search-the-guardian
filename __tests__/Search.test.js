import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {TextInput, TouchableOpacity} from 'react-native';

import Search from '../components/Search.js';

let fakeHandleChange;
let fakeGetArticles;
let props;

describe('Search Component', () => {
  beforeEach(() => {
    fakeHandleChange = jest.fn();
    fakeGetArticles = jest.fn();
    props = {
      handleChange: fakeHandleChange,
      searchInput: '',
      getArticles: fakeGetArticles,
    };
  });

  test('Snapshot - it renders correctly', () => {
    const snapshot = renderer.create(<Search {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('Search input field: calls handleChange function when text changes', () => {
    const fakeSearch = renderer.create(<Search {...props} />);
    const inputField = fakeSearch.root.findByType(TextInput);
    inputField.props.onChangeText('cycling');
    expect(fakeHandleChange).toHaveBeenCalledWith('cycling');
  });

  test('Go! button: calls getArticles function when pressed', () => {
    props.searchInput = 'cycling';
    const fakeSearch = renderer.create(<Search {...props} />);
    const button = fakeSearch.root.findByType(TouchableOpacity);
    button.props.onPress();
    expect(fakeGetArticles).toHaveBeenCalled();
  });
});
