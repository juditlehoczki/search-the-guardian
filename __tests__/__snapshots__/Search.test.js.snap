// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Snapshot - it renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 20,
        "marginLeft": 5,
      }
    }
  >
    🔍
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search..."
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#D1529E",
        "borderRadius": 5,
        "borderStyle": "dotted",
        "borderWidth": 1,
        "color": "#0F406E",
        "flex": 8,
        "marginLeft": 5,
        "marginVertical": 5,
        "padding": 15,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D1529E",
        "borderRadius": 15,
        "flex": 1,
        "justifyContent": "center",
        "margin": 5,
        "opacity": 1,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Go!
    </Text>
  </View>
</View>
`;
